pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/writercallable/WriterCallableFactory.java

```diff
@@ -75,16 +75,15 @@ public void setQuery(String query) {
     /**
      * Set batch size to use.
      *
-     * @param batchSize < 0: Use batches of infinite size
+     * @param batchSize = 0: Use batches of recommended size
      * @param batchSize = 1: Do not use batches
      * @param batchSize > 1: Use batches of the given size
+     * @param batchSize < 0: Use batches of infinite size
      */
     public void setBatchSize(int batchSize) {
-        if (batchSize < 0) {
-            batchSize = 0;
-        }
-        else if (batchSize == 0) {
-            batchSize = 1;
+        if (batchSize == 0) {
+            // Set the recommended value: https://docs.oracle.com/cd/E11882_01/java.112/e16548/oraperf.htm#JJDBC28754
+            batchSize = 100;
```

Comment:
this is counter-intuitive, a good guideline is to never change the meaning of the user-set parameter. I would propose the following -- user provides value of BATCH_SIZE:
1. nothing (not provided) -- use default of 100
2. 0 -- error out -- batch size of 0 does not make sense
3. 1 -- error out -- batch size of 1 does not make sense
4.  < 1 -- error out -- batch size of <1 does not make sense
5. > 1 and < 65k -- use the number provided by the user
6. > 65k -- error out -- too large for practical purposes ?

Reply:
This makes sense.
Do you propose to use batches by default? Currently, `BATCH_SIZE` controls not only the size of a batch, but also whether it is used or not.
Also, if we allow user not to use batches, we should make `1` (or some other value) a valid `BATCH_SIZE`.

Reply:
oh, thanks, got it, I thought batch on/off was a separate property. If we want to stick with a single property (and keep it simple for users), then I think using batching by default (with a default value) is a good idea (if the backend db supports it). This way, though, we will have to let users to turn it off for a query, which they will achieve by setting the value to 0 (batch feature is off) or 1 (batch with size 1 practically means no batching, implementation can take care of that). I think this is closer to your original design, except I would take out negative values to prohibit unlimited size, as well as maybe put an upper bound on batch size to prevent crashing PXF with OOM.

Reply:
As I understand, now optimal parameter values look like this:
```
1. nothing (not provided) -- use default of 100
2. 0 -- no batching
3. 1 -- no batching
4. >1 -- use the number provided by the user
```
In my opinion, we shouldn't limit possible `BATCH_SIZE` max value (or we should set it very high, to 500k for example) as we dont know the usecases and external DBs that users will use it with. It is possible that there are some DBs or other systems that use high batch size for optimal inserts.
@denalex what do you think about `BATCH_SIZE` max value?

Reply:
@kapustor -- agree, limiting max is superficial, as with a given memory there might be a value below max that would cause OOM and there is no other reasons to limit the max. So, let's not limit the max and just use a value provided by the user.